File: angular/AirNstay/Backend/src/main/java/com/cognizant/demo/entity/SupportTicket.java
package com.cognizant.demo.entity;

import java.util.Date;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class SupportTicket {

    @Id
    private String ticketId;

    private String userID;
    private String name;
    private String detailedIssue;

    @Temporal(TemporalType.TIMESTAMP)
    private Date date;

    private String status = "Pending";
    private String agent = "";
    private boolean agentAssigned = false;
    private boolean completed = false;
}

File: angular/AirNstay/Frontend/src/app/SupportTicket.ts
export default class SupportTicket{
    ticketId! : string;
    userID : string | null;
    name! : string;
    detailedIssue! : string;
    date! : Date;
    status! : string;
    agent! : string;
    agentAssigned!: boolean;
    completed!: boolean;

    constructor(ticketId : string,userID :string | null, name : string, detailedIssue : string,date : Date)
    {
        this.ticketId=ticketId;
        this.userID=userID;
        this.name=name;
        this.detailedIssue=detailedIssue;
        this.date=date;
        this.status="Pending";
        this.agent="";
        this.agentAssigned = false;
        this.completed = false;
    }
}